### PR TITLE
Fix __getattr__ in WorkerBase

### PR DESCRIPTION
**Pull request by Czaki**
# Description

`QRunnable` does not contain `__getattr__` function.

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# References

found in #3366

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

File: napari/_qt/qthreading.py [hunk 1]
```diff
@@ -97,7 +97,9 @@ def __getattr__(self, name):
             # (which is of type `SignalInstance` in PySide and
             # `pyqtBoundSignal` in PyQt)
             return getattr(self.signals, name)
-        return super().__getattr__(name)
+        raise AttributeError(
+            f"{self.__class__.__name__} has no attribute {name}"
```
Comment (tlambert03) [inline napari/_qt/qthreading.py]:
```suggestion
            f"{self.__class__.__name__!r} object has no attribute {name!r}"
```

**Review by tlambert03 — APPROVED** [review]
thanks for catching!

**Comment by Czaki** [comment]
@tlambert03 why do you use `!r` (representation) on strings?

**Comment by tlambert03** [comment]
> @tlambert03 why do you use !r (representation) on strings?

so that it gets wrapped in quotes, which makes it clearer that the string is not a part of the error message itself, but rather a variable that you have injected into the message.  (and to match the standard library:)

```py
In [1]: obj = object()

In [2]: obj.asdfg
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-2-d397fdd372af> in <module>
----> 1 obj.asdfg

AttributeError: 'object' object has no attribute 'asdfg'
```

I prefer that error message to:

```py
 AttributeError: object object has no attribute asdfg
```

**Comment by tlambert03** [comment]
> > @tlambert03 why do you use !r (representation) on strings?
> 
> so that it gets wrapped in quotes, which makes it clearer that the string is not a part of the error message itself, but rather a variable that you have injected into the message. (and to match the standard library:)
> 
> ```python
> In [1]: obj = object()
> 
> In [2]: obj.asdfg
> ---------------------------------------------------------------------------
> AttributeError                            Traceback (most recent call last)
> <ipython-input-2-d397fdd372af> in <module>
> ----> 1 obj.asdfg
> 
> AttributeError: 'object' object has no attribute 'asdfg'
> ```

I prefer that error message to:

```py
> 
> AttributeError: object object has no attribute asdfg
> 
```

**Comment by Czaki** [comment]
I do not remember about this feature fo `!r`.

**Comment by tlambert03** [comment]
> I do not remember about this feature fo `!r`.

yep, it's a nice trick.  the `repr` of a string includes the quote:

```python
In [1]: repr("hi")
Out[1]: "'hi'"
```

**Review by jni — APPROVED** [review]

**Pull request merged**